diff --git a/03/Android/MySQLDataApp/app/src/main/java/com/example/mysqldataapp/data/source/Source.kt b/03/Android/MySQLDataApp/app/src/main/java/com/example/mysqldataapp/data/source/Source.kt
index 9df96f3..25c9b7c 100644
--- a/03/Android/MySQLDataApp/app/src/main/java/com/example/mysqldataapp/data/source/Source.kt
+++ b/03/Android/MySQLDataApp/app/src/main/java/com/example/mysqldataapp/data/source/Source.kt
@@ -27,6 +27,7 @@ data class PersonEntity(
     @PrimaryKey(autoGenerate = true) val id: Int = 0,
     @ColumnInfo(name = "name") val name: String?,
     @ColumnInfo(name = "surname") val surname: String?,
+    @ColumnInfo(name = "language") val language: String?,
 )
 
 // Data access object
diff --git a/03/Android/MySQLDataApp/app/src/main/java/com/example/mysqldataapp/ui/state/State.kt b/03/Android/MySQLDataApp/app/src/main/java/com/example/mysqldataapp/ui/state/State.kt
index f288111..25bad1c 100644
--- a/03/Android/MySQLDataApp/app/src/main/java/com/example/mysqldataapp/ui/state/State.kt
+++ b/03/Android/MySQLDataApp/app/src/main/java/com/example/mysqldataapp/ui/state/State.kt
@@ -32,12 +32,14 @@ data class Person(
 fun Person.toEntity() = PersonEntity(
     id = id,
     name = name,
-    surname = surname)
+    surname = surname,
+    language = language)
 
 fun PersonEntity.toPerson() = Person(
     id = id,
-    name = name!!, // TODO null safety
-    surname = surname!!)
+    name = name ?: "",
+    surname = surname ?: "",
+    language = language ?: "")
 
 // List ViewModel
 
